src/readsbml.jl

```diff
@@ -495,13 +495,40 @@ function extract_model(mdl::VPtr)::SBML.Model
         events[evname] = SBML.Event(evname, trig_math, event_assignments)
     end
 
+    # Rules
+    rules = Rule[]
+    num_rules = ccall(sbml(:Model_getNumRules), Cuint, (VPtr,), mdl)
+    for n in 0:(num_rules - 1)
+        rule_ptr = ccall(sbml(:Model_getRule), VPtr, (VPtr, Cuint), mdl, n)
+        type = if ccall(sbml(:Rule_isAlgebraic), Bool, (VPtr,), rule_ptr)
+            AlgebraicRule
+        elseif ccall(sbml(:Rule_isAssignment), Bool, (VPtr,), rule_ptr)
+            AssignmentRule
+        elseif ccall(sbml(:Rule_isRate), Bool, (VPtr,), rule_ptr)
+            RateRule
+        end
```

Comment:
I wonder if this should have an `else` branch throwing an error, in case we get a weird (well, wrong) model with an unknown rule type?

Reply:
well not an error but can we have a warning about unsupported contents (or just ignore it because SBML says there's no more assignment types?)

Reply:
Ok, well, at the moment `type` would be undefined if all branches are false, which is close to the "just ignore it" option.  I'm not even sure that's possible to have a rule not of one of those types, I suspect libsbml would skip it.

Reply:
Yeah, if I change the name of one of the rules to `myRule` I get:
```
┌ Error: SBML reported error: Apart from the general Notes and Annotation subobjects permitted on all SBML components, a ListOfRules container object may only contain Rule objects.
│ Reference: L3V1 Section 4.2.8
│  Element 'myRule' is not part of the definition of <listOfRules>.
└ @ SBML ~/.julia/dev/SBML/src/utils.jl:271
ERROR: AssertionError: Opening SBML document has reported errors
```
So probably that isn't a concern in practice, libsbml would fail much earlier

Reply:
ok that sounds good. :]